dvl-a50/dvl.py

```diff
@@ -175,12 +194,44 @@ def set_should_send(self, should_send):
         self.should_send = should_send
         self.save_settings()
 
+    def longitude_scale(self, lat):
+        scale = math.cos(lat * (math.pi / 180.0))
+        return max(scale, 0.01)
+
+    def latLngToXY (self, lat, lon):
+        x = (lat-self.origin[0]) * LATLON_TO_CM
+        y =  self.longitude_scale((lat + self.origin[0])/2) * LATLON_TO_CM * (lon-self.origin[1])
+        return [x, y]
+
+    def has_origin_set(self):
+        data = json.loads(self.mav.get("/GLOBAL_POSITION_INT/message"))
+        return data['lat'] != 0 or data['lon'] != 0
+
+    def set_current_position(self, lat, lon):
+        """
+        Sets the EKF origin to lat, lon
+        """
+        lat = float(lat)
+        lon = float(lon)
+        if not self.has_origin_set():
+            self.mav.set_gps_origin(lat, lon)
+            self.origin = [lat, lon]
+            self.save_settings()
+        else:
+            x, y = self.latLngToXY (lat, lon)
+            depth = float(self.mav.get("/VFR_HUD/message/alt"))
+
+            attitude = json.loads(self.mav.get("/ATTITUDE"))
+            positions = [x, y, -depth]
+            self.reset_counter += 1
+            self.mav.send_vision_position_estimate(self.timestamp, positions, reset_counter=self.reset_counter)
```

Comment:
Noting that `handle_position_local` is incomplete at the moment, because if the user turns on `POSITION_ESTIMATE` we'll be reading the values and passing them to the autopilot, but:
- we currently aren't using the "Reset Dead Reckoning" command (it should probably happen every time we set current position, and also make use of a stored position offset that gets applied to add on the current position (as a local offset from the GPS origin) to the dead reckoning one)
- we aren't enabling DVL "Gyro Calibration" (which should presumably happen when the vehicle gets launched)
- we aren't allowing a `mounting_rotation_offset` to be specified (which should be set when mounting, if not forward-aligned with the vehicle)

If we're going to properly enable that, we may as well add in the additional "Configuration parameters" at the same time, which would allow the user to set speed of sound, turn off the acoustics when out of water, turn off the DVL's blinking LED, and change the range mode to a manual option.

If it won't be supported in this release then I'd recommend removing the `POSITION_ESTIMATE` option from the frontend, and raising an issue about it as a TODO :-)

Reply:
yeah, it was introduced as kinda of an experimental thing that I left there, I'll play with it, but I'm leaning to hide it from end-users for now...